Implement yaml config file for downloading experimental data

jdmulligan opened the pull request:
@FHead @Soltz Here is an implementation of the configuration machinery.

The main steering script is `get_expdata.py`, which should be run as
```
python get_expdata.py -c expdata_config.yaml
```
where `expdata_config.yaml` is a user-defined configuration file, an example of which is provided at `expdata_config_sample.yaml`. You can specify one or more blocks in the config file for HEPData, PHENIX, or extractError or any combination, and they will all be processed according to their appropriate scripts. See the script and sample config file for further documentation. 

Please let me know what you think! 

(Also: I tested that it runs successfully for downloading the data, however for edit_data.py I got a crash due to the fact that the downloaded ATLAS errors contain "%" in them -- but I think this is unrelated to the configuration machinery)

jdmulligan committed:
Implement yaml config file for downloading experimental data

soltz commented:
James,

This is more or less what I had in mind with one block per data entry.  The fact the script can also read multiple entries looks useful, as it can simplify the config files when we want to read in multiple (or all) centrality values for a single measurement.

I'm hesitant to weigh in on any style discussions between you and Yi, as I believe any such opinions should be weighted by inverse age.  You're approach looks good to me, but I will also be OK with any approach that you and Yi agree upon.

-Ron

FHead commented on src/get_expdata.py:
```diff
@@ -0,0 +1,56 @@
+import get_hepdata
+import get_phenix
+import edit_data
+
```
```suggestion
import sys
```

FHead commented on src/get_hepdata.py:
```diff
@@ -67,54 +72,55 @@
+          e = str(err['symerror'])
+          # Convert if percentage
+          # if (e[-1]=='%'):
+          #   e = str(float(e[:-1])*v['value']*0.01)
```
Let's enable this part for now - we would avoid the atlas error you observed.  We can always come back and revisit this particular implementation if there are some fringe cases missed by this

FHead commented on src/get_expdata.py:
```diff
@@ -0,0 +1,56 @@
+    elif 'phenix' in entry:
+      print('Getting phenix data for entry "{}"...'.format(entry))
+      get_phenix.get_phenix(config[entry])
+    elif 'extractError' in entry:
```
Suggest two for loops: the first one looking for hepdata and phenix, and then after all those are done, and the second for loop targeting extractError.  The advantage for this is that the extractError block may not always be in the bottom

FHead replied:
Actually, is it possible to control extract error from the main config block?

Something like

```
hepdata:
   ...
   ...
   extractError: [False, False, True, True]
```

Then one doesn't need to type out the file name "Data_ATLAS_PbPb2760_RAACharged_30to40_2015.dat" which could be less error-prone - it can be automatically determined.  What do you think?

FHead replied:
Hmm but then it's a bit of a mess because there are other parameters to specify.  Not sure what is the best way for this, let's just change the for-loop for now if you don't have a good alternative.

jdmulligan replied:
Yea, indeed it's a bit safer to implement a second `for` loop, I'll implement that. In principle can add the extractError to the main config block as you suggest (probably with another layer of nesting), but yea I wonder if it may be a bit too fancy for the benefit...it would require a bit more structure to be imposed, and take away some of the flexibility that is there now (despite a bit of redundancy currently required).

FHead replied:
Yes, let’s do like this for now until we have a better alternative

FHead commented on src/expdata_config_sample.yaml:
```diff
@@ -0,0 +1,66 @@
+# provide separate config entries for one or more hepdata entries (as in "hepdata_CMS_HadronRAA_010" block).
+#-----------------------------------------------------------------------------------------------
+hepdata:
+  filepath: '../input/MATTERLBT1-test/'
```
After some thoughts, perhaps it's easier for users if we run the script at the base level (instead of going into the src folder).  So we could remove the "../" in the beginning in the example.

jdmulligan replied:
Sure, no preference from my side

FHead commented:
One additional suggestion on the expdata_config_sample.yaml file - shall we make a folder and put the file in there?

jdmulligan committed:
Implement several minor improvements to data downloading configuration machinery

jdmulligan commented:
Hi Yi Chen,
I implemented your suggestions -- I also moved the downloading scripts into a new folder separate from src -- not quite what you were suggesting, but I think these scripts are pretty much self-contained and unrelated to src, so it's nicer to de-clustter src. But feel free to re-organize in any way you prefer, no strong opinion from me :)

FHead commented:
Hi James, thank you for the updates.  I will merge the PR now - I would prefer the scripts to be in the src directory, but we can update that later.
